pkg/util/refreshable/refreshable.go

```diff
@@ -63,13 +64,16 @@ func (a *authorizer) RefreshWithContext(ctx context.Context, log *logrus.Entry)
 			azureerrors.IsInvalidSecretError(err) {
 			return false, nil
 		}
-
 		return false, err
 	}
 
 	return true, nil
 }
 
+func (a *authorizer) RefreshWithContextRawError(ctx context.Context) error {
+	return a.sp.RefreshWithContext(ctx)
+}
+
```

Comment:
I stand by my previous comments, I don't think this should be separated from the other method. we should reduce the scope of the original authorizer.RefreshWithContext and add a function to go from `error` to `bool, error`. 

Tbh I think the whole interface is useless because we only use simple functions from the `*adal.ServicePrincipalToken` and we could probably just use that instead. not necessarily your doing but i'd rather not pile up and add more methods to something like this.

Reply:
@facchettos this change was made [at Mikalai's request](https://github.com/Azure/ARO-RP/pull/2202#discussion_r927624201) and is the simplest way to achieve this without breaking the existing pattern. I agree completely that this needs to be refactored, but can be done in a follow up where we address the overall design.

Reply:
I stand by my comments. `RefreshWithContext` has the same name than `sp.RefreshWithContext` but has a different logic, while `RefreshWithContextRawError` has a different name but the same logic than `sp.RefreshWithContext`

having  it this way would make more sense:

```go
func (a *authorizer) RefreshWithContext(ctx context.Context, log *logrus.Entry) (error) {
	return a.sp.RefreshWithContext(ctx)
}
```

```go 
func RefreshIsDone(log *logrus.Entry, err error) (bool, error) {
	if err != nil {
		log.Info(err)

		if !autorest.IsTokenRefreshError(err) ||
			azureerrors.IsUnauthorizedClientError(err) ||
			azureerrors.IsInvalidSecretError(err) {
			return false, nil
		}
		return false, err
	}

	return true, nil
}
```

Reply:
then we don't need to have the `RefreshWithContextRawError` method on the interface, which is making it more complicated than it has to be

Reply:
Once this is done, we should be able to do the changes that I outlined below without further modifications

Reply:
```go
func RefreshWithContextOldLogic(a Authorizer, ctx context.Context) (book, error) {
    err := a.RefreshWithContext(ctx)
    return RefreshIsDone(err)
}
``` 
to avoid duplication where the current `a.RefreshWithContext` is used